### PR TITLE
feat: use throttled punctuator

### PR DESCRIPTION
**Pull request by kishansairam9**

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ hypertrace-entity-service = "0.8.78"
 hypertrace-config-service = "0.1.54"
 hypertrace-grpc-utils = "0.12.4"
 hypertrace-serviceFramework = "0.1.60"
-hypertrace-kafkaStreams = "0.3.8"
+hypertrace-kafkaStreams = "0.4.0-SNAPSHOT"
```
Comment (kishansairam9) [inline gradle/libs.versions.toml:6]: depends on https://github.com/hypertrace/kafka-streams-framework/pull/82

File: raw-spans-grouper/raw-spans-grouper/src/main/resources/configs/common/application.conf [hunk 2]
```diff
@@ -61,3 +61,12 @@ metrics.reporter {
...
 dataflow.metriccollection.sampling.percent = 10.0
+
+trace.emit.callback.registry {
+  frequency = 1s
+  frequency = ${?TRACE_EMIT_CALLBACK_REGISTRY_FREQUENCY}
+  yield = 5s
+  yield = ${?TRACE_EMIT_CALLBACK_REGISTRY_YIELD}
+  window = 5ms
+  window = ${?TRACE_EMIT_CALLBACK_REGISTRY_WINDOW}
+}
```
Comment (kishansairam9) [inline raw-spans-grouper/raw-spans-grouper/src/main/resources/configs/common/application.conf:72]: these defaults require some experimental tuning

File: raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/TraceEmitTasksPunctuator.java [hunk 3]
```diff
@@ -70,24 +67,24 @@ class TraceEmitPunctuator implements Punctuator {
...
   private final ProcessorContext context;
   private final KeyValueStore<SpanIdentity, RawSpan> spanStore;
   private final KeyValueStore<TraceIdentity, TraceState> traceStateStore;
   private final To outputTopicProducer;
   private final long groupingWindowTimeoutMs;
-  private Cancellable cancellable;
 
-  TraceEmitPunctuator(
-      TraceIdentity key,
+  TraceEmitTasksPunctuator(
```
Comment (kishansairam9) [inline raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/TraceEmitTasksPunctuator.java:78]: We observed for some reason, the bytes read & written compaction rate has increased for span-data-store after this PR. Still looking into why it is happening, hopefully we can fix it

Comment (laxmanchekka) [reply raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/TraceEmitTasksPunctuator.java]: > the bytes read & written compaction rate has increased 
False alarm. No increase in actual iops.

**Review by laxmanchekka — APPROVED** [review]

**Review by laxmanchekka — DISMISSED** [review]

**Review by laxmanchekka — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]
Reviewed test case issue, rest is already reviewed by Laxman.

**Pull request merged**